subsys/bootloader/mcuboot_image_header_parser/mcuboot_image_header_parser.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr/types.h>
+#include <string.h>
+#include <image.h>
+#include <errno.h>
+
+/**@brief Reads MCUBoots Image header
+ *
+ * @param header_addr Address of where the image header is located.
+ * @param[out] image_header Pointer to the image header struct which should be
+ * populated.
+ *
+ * @retval 0 If successfully read the IMAGE_HEADER magic meaning an MCUBoot
+ * header was present.
+ * @retval -EINVAL If the image_header magic doesn't match meaning an MCUBoot
+ * header was not found at the given address.
+ */
+static int mcuboot_read_img_hdr(u32_t *header_addr,
+				struct image_header *image_header)
+{
+	memcpy(image_header, header_addr, IMAGE_HEADER_SIZE);
+	if (image_header->ih_magic != IMAGE_MAGIC) {
+		return -EINVAL;
+	}
+	return 0;
+}
+
+uint32_t mcuboot_read_version_number(u32_t *header_addr)
+{
+	struct image_header img_hdr;
+	int found_header = mcuboot_read_img_hdr(header_addr, &img_hdr);
```

Comment:
It it necessary to read out the header using `memcpy`? Could it be done like

```
struct image_header * img_hdr = (struct image_header *)header_addr;

if (img_hdr->ih_magic != MAGIC) {
  return -ESOMETHING;
}

...
```

Reply:
Compiles down to the same code but yes it is possible.